diff --git a/UbuntuARM.min b/UbuntuARM.min
old mode 100644
new mode 100755
index 2e7365a..b18f51e
--- a/UbuntuARM.min
+++ b/UbuntuARM.min
@@ -1,7 +1,17 @@
 # stand-alone executable
-
+BUILD_EXES+=graph_app
 graph_app_QAICIDLS += interface/hexagon_nn \
                         $(DSPCV_DIR)/inc/dspCV
+
+# stub library
+BUILD_DLLS += libhexagon_nn
+libhexagon_nn_QAICIDLS += interface/hexagon_nn \
+                        $(DSPCV_DIR)/inc/dspCV
+libhexagon_nn_C_SRCS += $(V)/hexagon_nn_stub \
+         $(V)/dspCV_stub \
+         test/graphinfo \
+         $(COMPILE_GRAPHINIT)
+
 ifndef GRAPHINIT
 GRAPHINIT=test/graphinit_small
 endif
@@ -17,14 +27,16 @@ graph_app_C_SRCS += \
     test/append_const_node_large_array \
     $(COMPILE_GRAPHINIT) \
     $(V)/hexagon_nn_stub \
-	$(V)/dspCV_stub 
+   $(V)/dspCV_stub
 
 graph_app_C_SRCS += $(TESTDATA:.c=)
 
 ifeq ($(CDSP_FLAG), 1)
-	graph_app_DLLS += libcdsprpc
+   graph_app_DLLS += libcdsprpc
+   libhexagon_nn_DLLS += libcdsprpc
 else
-	graph_app_DLLS += libadsprpc
+   graph_app_DLLS += libadsprpc
+   libhexagon_nn_DLLS += libadsprpc
 endif
 graph_app_DEFINES += VERIFY_PRINT_ERROR
 
@@ -36,5 +48,5 @@ BUILD_COPIES = \
    $(DLLS) \
    $(EXES) \
    $(LIBS) \
-   $(SHIP_DIR)/ 
+   $(SHIP_DIR)/
 
